stepper control added

prat-degwekar opened the pull request:
with minimal changes to the main code, so that new changes need to be just commented out.
also, specific, bot and board based small edits need to be made, but they can be done very easily...

prat-degwekar committed:
Added stepper usability

to use stepper, find proper conversion value first, then call angleMovement, small edits left though...

prat-degwekar committed:
Updates made

At this point, ideally, steppers should work.

reisub0 commented:
@prat-degwekar 

All code in this repo has been formatted, and if you wish to contribute, you'll have to format your code with AutoPEP8.

Also, it would be a much better idea to have a boolean param stepper = True so the constructor would avoid initialising all of these attributes if it is a stepper, then it can be merged in the main motor class itself.

If you're importing motor from this module, then there's no need to keep around all of the cw and ccw attribs as it's not a DC motor anymore.

Also you haven't even gotten the parameters from the user. Try interpreting the code on an online compiler. This code would just give an exception because the values of pin1, etc. aren't even defined.

prat-degwekar committed:
Updated requirements to PEP8 format

logic updated, with removed cw, ccw implementations for a DC Motor.

prat-degwekar committed:
Stepper class completed

Integration of stepper and motor on the way.

prat-degwekar committed:
cleaned up

cleaner and more integrable code

prat-degwekar committed:
Added stop code

prat-degwekar committed:
Class that combines both Motor and Stepper

Correctivity needs to be checked, very specifically wrt object wheeler, or needs to be seen how that functionality is to be completed. Otherwise, Wheel class encapsulates motor and stepper, so as to keep them separate, yet together...

prat-degwekar commented:
Updated things as requested, but not sure if wheeler object works, just check that, or see how that function can be implemented. #1
